Src/VimWpf/Implementation/Misc/KeyMappingTimeoutHandler.cs

```diff
@@ -80,6 +80,7 @@ private void OnKeyInputBuffered(object sender, KeyInputSetEventArgs args)
                     if (!globalSettings.Timeout)
                     {
                         _timer.Stop();
+                        return;

```

Comment:
That comment above this even explicitly says "and return". Not sure how we missed this originally. Thanks for finding this!